src/main/java/seedu/address/model/date/Date.java

```diff
@@ -59,21 +59,33 @@ public static boolean isDateTodayOrAfter(LocalDate date) {
     }
 
     /**
-     * Returns true if a given date is today or after today.
+     * Compares two EventDates by date.
+     * @param secondEventDate second EventDate to compare
+     * @return -1 if first EventDate is before second EventDate, 0 if equal, 1 if first EventDate is after
+     */
+    public int compareTo(Date secondEventDate) {
+        LocalDate firstDate;
+        LocalDate secondDate;
+
+        firstDate = LocalDate.parse(this.date);
+        secondDate = LocalDate.parse(secondEventDate.date);
+
+        return firstDate.compareTo(secondDate);
+    }
+
+    /**
+     * Returns true if the date is within 3 days from today.
+     * @return true if date is within 3 days from today, false, otherwise.
      */
-    public boolean isDueWithinThreeDays() {
+    public boolean isWithinThreeDays() {
         LocalDate today = LocalDate.now();
-        LocalDate date = LocalDate.parse(this.date);
-        if (date.isBefore(today)) {
-            return true;
-        }
-        if (date.isEqual(today)) {
-            return true;
-        }
-        if (date.isAfter(today) && date.isBefore(today.plusDays(3))) {
+        LocalDate taskDue = LocalDate.parse(this.date);
+
+        if (taskDue.isBefore(today.plusDays(4))) {
```

Comment:
in the spirit of abstraction perhaps `today.plusDays(4)` can be defined as a static variable in the Date class to follow the format of the codebase

Reply:
Decided to create a new method to encompass the getting of the date four days after today.